src/backend/distributed/planner/multi_logical_optimizer.c

```diff
@@ -4102,40 +4130,56 @@ WorkerLimitCount(MultiExtendedOp *originalOpNode)
 	 * aggregates, we can push down the original limit. Else if we have order by
 	 * clauses with commutative aggregates, we can push down approximate limits.
 	 */
-	if (groupClauseList == NIL)
+	if (groupClauseList == NIL || sortClauseList == NIL || !hasOrderByAggregate)
```

Comment:
> We have discussed this before. The problem here is that, we do not specify any order and there is no guarantee on the result set if no order-by is specified.

I think we discussed this if there is not any aggregate in the query like you noted in #606, then it makes sense. But, if there is an aggregate, we can return the wrong aggregate value. Does this make sense?